### PR TITLE
Convert2react

### PR DESCRIPTION
**Pull request by dhagandev**

### CODE REVIEW

File: src/js/components.js [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    componentDidMount() {
+        this.apiCall();
+    }
+
+    handler(e) {
+        this.setState({
+            data: null,
+            error: null
+        });
+        e.preventDefault();
```
Comment (kev-chan) [inline src/js/components.js]: why preventDefault() ?

File: src/js/components.js [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        this.apiCall();
+    }
+
+    apiCall() {
```
Comment (kev-chan) [inline src/js/components.js]: should the api call be in this class?

File: src/js/components.js [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+            }
+        );  
+    }
+
+    render() {
+        let error = this.state.error;
+        let data = this.state.data;
+        let returnElement = null;
+
+        if (error == null && data == null) {
```
Comment (kev-chan) [inline src/js/components.js]: why only if statements?

File: src/js/components.js [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+class TweetInfoComponent extends React.Component {
+    render() {
+        let fullDate = new Date(this.props.tweet.createdAt);
+        let dateToShow = month[fullDate.getMonth()].substring(0, 3) + " " + fullDate.getDate();
+
+        return create(
+            "div",
+            {
+                key: "tweetInfo" + this.props.index,
```
Comment (kev-chan) [inline src/js/components.js:186]: why is there a key on all 'create' calls?

File: src/js/components.js [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+            data: null,
+            error: null
+        });
+        e.preventDefault();
+        this.apiCall();
+    }
+
+    apiCall() {
+        let apiUrl = "http://" + serverHostname + ":" + serverPort + "/" + timelinePath;
+        fetch(apiUrl)
```
Comment (kev-chan) [inline src/js/components.js]: why the switch for how the API call is made?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/js/reactRenderComponents.js [hunk 6]
```diff
@@ -1,10 +1,10 @@
-import {HelloComponent} from './hello.js';
+import {HeaderComponent, BodyComponent} from './components.js';
 import React from 'react';
 import ReactDOM from 'react-dom';
 
 export const run = () => {
     ReactDOM.render(
-        React.createElement(HelloComponent),
-        document.getElementById("reactComponent")
+        [React.createElement(HeaderComponent, {key: "headerComponent"}, null), React.createElement(BodyComponent, {key: "bodyComponent"}, null)],
```
Comment (kev-chan) [inline src/js/reactRenderComponents.js]: why the passing of null?

File: src/scss/styles.scss [hunk 7]
```diff
@@ -21,88 +21,92 @@
...
+            @include margins(5px 0px);
+            @include fontSizeAlign(24px, center);
+            color: #ff0000;
+        }
 
-            .userImg {
-                @include margins(10px 0px 4px);
-                border-radius: 50%;
-            }
+        .tweetRow:nth-child(odd) {
```
Comment (kev-chan) [inline src/scss/styles.scss]: use scss for these rules

File: src/js/components.js (hunk 1 above)
Comment (dhagandev) [reply src/js/components.js]: Removed.

File: src/js/components.js (hunk 2 above)
Comment (dhagandev) [reply src/js/components.js]: I originally had it in the timeline table, but I had no good way of telling the table it needed to make the api call again and update itself when the button was pressed.

File: src/js/components.js (hunk 5 above)
Comment (dhagandev) [reply src/js/components.js]: I don't understand what you are asking?

File: src/js/components.js (hunk 4 above)
Comment (dhagandev) [reply src/js/components.js:186]: Otherwise it would throw warnings and clutter the console.

File: src/js/components.js (hunk 3 above)
Comment (dhagandev) [reply src/js/components.js]: Converted to if/elses.

File: src/js/reactRenderComponents.js (hunk 6 above)
Comment (dhagandev) [reply src/js/reactRenderComponents.js]: The null element represents their children, the children are handled in the Components. This allows for the Dom Render code to be cleaner, I don't need to import/export each component

Comment (dhagandev) [reply src/js/reactRenderComponents.js]: Removed nulls.

File: src/js/components.js [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+    render() {
+        return create(
+            "div",
+            {key: "header", className: "header"},
+            "Lab for Dana"
+        );
+    }
+}
+
+export class BodyComponent extends React.Component {
```
Comment (kev-chan) [inline src/js/components.js:17]: use better variable names overall for handler/getApi

File: src/js/components.js [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+        };
+        this.getApi = this.getApi.bind(this);
+    }
+
+    componentDidMount() {
+        apiCall(this.getApi);
+    }
+
+    handler() {
+        this.setState({
```
Comment (kev-chan) [inline src/js/components.js]: why did you decide to change the state here?

File: src/js/main.js [hunk 10]
```diff
@@ -1,15 +1,9 @@
-import {getTimeline} from './data.js';
 import {run} from './reactRenderComponents.js';
 const hostname = '127.0.0.1';
 const port = 9000;
 
 console.log(`Server running at http://${hostname}:${port}/`);
```
Comment (kev-chan) [inline src/js/main.js]: do you still need the console.log?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/js/components.js (hunk 9 above)
Comment (dhagandev) [reply src/js/components.js]: By setting data and error to null here it triggers the waiting screen if it is being slow to receive the api update.

File: src/js/main.js (hunk 10 above)
Comment (dhagandev) [reply src/js/main.js]: I thought it was good information to have but I can remove it if you feel it isn't necessary.

File: src/js/components.js (hunk 8 above)
Comment (dhagandev) [reply src/js/components.js:17]: Done

File: src/js/main.js [hunk 11]
```diff
@@ -1,15 +1,5 @@
-import {getTimeline} from './data.js';
 import {run} from './reactRenderComponents.js';
```
Comment (kev-chan) [inline src/js/main.js]: why have main.js only call run?

File: src/js/components.js [hunk 12]
```diff
@@ -0,0 +1,217 @@
...
+        this.onTimelineClickHandler = this.onTimelineClickHandler.bind(this);
+        this.state = {
+            data: null,
+            error: null
+        };
+        this.processApiCallResults = this.processApiCallResults.bind(this);
+    }
+
+    componentDidMount() {
+        apiCall(this.processApiCallResults);
```
Comment (kev-chan) [inline src/js/components.js]: better names for both of these calls

File: src/js/components.js [hunk 13]
```diff
@@ -0,0 +1,217 @@
...
+
+export class BodyComponent extends React.Component {
+    constructor(props) {
+        super(props);
+        this.onTimelineClickHandler = this.onTimelineClickHandler.bind(this);
+        this.state = {
+            data: null,
+            error: null
+        };
+        this.processApiCallResults = this.processApiCallResults.bind(this);
```
Comment (kev-chan) [inline src/js/components.js]: keep the binds together

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/js/components.js [hunk 14]
```diff
@@ -18,26 +18,26 @@ export class BodyComponent extends React.Component {
...
         this.state = {
             data: null,
             error: null
         };
-        this.processApiCallResults = this.processApiCallResults.bind(this);
     }
 
     componentDidMount() {
-        apiCall(this.processApiCallResults);
+        apiGetTimeline(this.processTimelineResults);
```
Comment (kev-chan) [inline src/js/components.js]: nice the names are much better now

**Review by kev-chan — APPROVED** [review]

**Pull request merged**